TutorLizard.BusinessLogic/Services/StudentService.cs

```diff
@@ -1,11 +1,41 @@
-﻿using TutorLizard.BusinessLogic.Interfaces.Services;
+﻿using Microsoft.EntityFrameworkCore;
+using TutorLizard.BusinessLogic.Interfaces.Data.Repositories;
+using TutorLizard.BusinessLogic.Interfaces.Services;
+using TutorLizard.BusinessLogic.Models;
+using TutorLizard.BusinessLogic.Models.DTOs;
+using TutorLizard.BusinessLogic.Models.DTOs.Requests;
+using TutorLizard.BusinessLogic.Models.DTOs.Responses;
 
 namespace TutorLizard.BusinessLogic.Services;
 public class StudentService : IStudentService
 {
-    // TODO inject needed repositories
-    public StudentService()
+    private readonly IDbRepository<Ad> _adRepository;
+    private readonly IDbRepository<AdRequest> _adRequestRepository;
+    public StudentService(IDbRepository<Ad> adRepository,
+                          IDbRepository<AdRequest> adRequestRepository)
     {
-        
+        _adRepository = adRepository;
+        _adRequestRepository = adRequestRepository;
+    }
+
+    public async Task<StudentsAcceptedAdsResponse> ViewAcceptedAds(StudentsAcceptedAdsRequest request)
+    {
+        var studentId = request.StudentId;
+
+        var acceptedAdRequests = await _adRequestRepository.GetAll()
+            .Where(ar =>  ar.StudentId == studentId && ar.IsAccepted)
+            .ToListAsync();
+
+        var acceptedAds = acceptedAdRequests
+            .Select(ar => ar.Ad).ToList();
+
+        var adListDtos = acceptedAds
+            .Select(ad => new AdListItemDto())
```

Comment:
You're not using `ad` when creating `AdListItemDto`, so it only has default values. You can use `{ }` to set its propetries' values based on `ad`'s.

Reply:
Added to the code.

Reply:
You missed 2 properties of `AdListItemDto`:
- CategoryName
- TutorName

You can access them via navigation properties (you included them for this very reason), e.g. `CategoryName = ad.Category.Name`